{{cookiecutter.project_slug}}/frontend/src/store/modules/main.ts

```diff
@@ -0,0 +1,252 @@
+    try {
+      const response = await api.logInGetToken(payload.username, payload.password);
+      const token = response.data.access_token;
+      if (token) {
+        saveLocalToken(token);
+        this.setToken(token);
+        this.setLoggedIn(true);
+        this.setLogInError(false);
+        await this.getUserProfile();
+        await this.routeLoggedIn();
+        this.addNotification({ content: "Logged in", color: "success" });
+      } else {
+        await this.logOut();
+      }
+    } catch (err) {
+      this.setLogInError(true);
+      await this.logOut();
+    }
+  }
+
+  @Action
+  async getUserProfile() {
+    try {
+      const response = await api.getMe(this.token);
+      if (response.data) {
+        this.setUserProfile(response.data);
+      }
+    } catch (error) {
+      await this.checkApiError(error);
+    }
+  }
+
+  @Action
+  async updateUserProfile(payload: IUserProfileUpdate) {
+    try {
+      const loadingNotification = { content: "saving", showProgress: true };
+      this.addNotification(loadingNotification);
+      const response = (
+        await Promise.all([
+          api.updateMe(this.token, payload),
+          await new Promise((resolve, _reject) => setTimeout(() => resolve(), 500)),
+        ])
+      )[0];
+      this.setUserProfile(response.data);
+      this.removeNotification(loadingNotification);
+      this.addNotification({
+        content: "Profile successfully updated",
+        color: "success",
+      });
+    } catch (error) {
+      await this.checkApiError(error);
+    }
+  }
+
+  @Action
+  async checkLoggedIn() {
+    if (!this.isLoggedIn) {
+      let token = this.token;
+      if (!token) {
+        const localToken = getLocalToken();
+        if (localToken) {
+          this.setToken(localToken);
+          token = localToken;
+        }
+      }
+      if (token) {
+        try {
+          const response = await api.getMe(token);
+          this.setLoggedIn(true);
+          this.setUserProfile(response.data);
+        } catch (error) {
+          await this.removeLogIn();
+        }
+      } else {
+        await this.removeLogIn();
+      }
+    }
+  }
+
+  @Action
+  async removeLogIn() {
+    removeLocalToken();
+    this.setToken("");
+    this.setLoggedIn(false);
+  }
+
+  @Action
+  async logOut() {
+    await this.removeLogIn();
+    await this.routeLogOut();
+  }
+
+  @Action
+  async userLogOut() {
+    await this.logOut();
+    this.addNotification({ content: "Logged out", color: "success" });
+  }
+
+  @Action
+  async routeLogOut() {
+    if (router.currentRoute.path !== "/login") {
+      router.push("/login");
+    }
+  }
+
+  @Action
+  async checkApiError(payload: AxiosError) {
+    if (payload.response && payload.response.status === UNAUTHORIZED) {
+      await this.logOut();
+    }
+  }
+
+  @Action
+  async routeLoggedIn() {
+    if (router.currentRoute.path === "/login" || router.currentRoute.path === "/") {
+      router.push("/main");
```

Comment:
This leads to an error (even though everything works fine):
```
Uncaught (in promise) Error: Redirected when going from "/login" to "/main" via a navigation guard.
```

Changing this to `/main/dashboard` would remove this error but would also make it more work to change the "main" page, as this line has to change too.

Reply:
This is also something I think could be addressed in the main branch as a general improvement.

Reply:
As with the other changes, I think including this would be good and I will also address this in a extra PR.

```suggestion
      router.push("/main/dashboard");
```